### PR TITLE
[BEAM-2074,BEAM-2073] Fixed SourceDStream's rate control usage.

### PR DESCRIPTION
**Pull request by staslev**
Be sure to do all of the following to help us incorporate your contribution
quickly and easily:

 - [x] Make sure the PR title is formatted like:
   `[BEAM-<Jira issue #>] Description of pull request`
 - [ ] Make sure tests pass via `mvn clean verify`. (Even better, enable
       Travis-CI on your fork and ensure the whole test matrix passes).
 - [x] Replace `<Jira issue #>` in the title with the actual Jira issue
       number, if there is one.
 - [ ] If this contribution is large, please file an Apache
       [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

---

### CODE REVIEW

**Comment by staslev** [comment]
Run Spark ValidatesRunner

**Comment by staslev** [comment]
retest this please.

**Comment by staslev** [comment]
Run Spark ValidatesRunner

**Comment by staslev** [comment]
R: @aviemzur

File: runners/spark/src/main/java/org/apache/beam/runners/spark/io/SourceDStream.java [hunk 1]
```diff
@@ -124,8 +124,34 @@
...
+  private long computeReadMaxRecords() {
+    if (boundMaxRecords > 0) {
+      LOG.info("Max records per batch has been set to {}, as configured in the PipelineOptions.",
+               boundMaxRecords);
+      return boundMaxRecords;
+    } else {
+      final scala.Option<Long> rateControlledMax = rateControlledMaxRecords();
+      if (rateControlledMax.isDefined()) {
+        LOG.info("Max records per batch has been set to {}, as advised by the rate controller.",
+                 rateControlledMax.get());
```
Comment (aviemzur) [inline runners/spark/src/main/java/org/apache/beam/runners/spark/io/SourceDStream.java]: Could extract param as this is called twice. not crucial.

**Comment by staslev** [comment]
Prior to submitting this PR I briefly played around with trying to test it, but could not find an easy way to test things at the level of inspecting rates per individual batches and the likes.

I believe this deserves a separate JIRA so please feel free to file one.

**Comment by staslev** [comment]
Rebased against #2698.

**Comment by staslev** [comment]
Run Spark ValidatesRunner

**Comment by staslev** [comment]
Run Spark ValidatesRunner

**Comment by staslev** [comment]
Run Spark ValidatesRunner

**Pull request merged**